### PR TITLE
feat: update the website application

### PR DESCRIPTION
**Pull request by nazarepiedady**
Hi, dear colleagues
I hope you are good,

In this pull request, I am applying some changes that will do an update to the official website just to follow the example given by the **vue-router** documentation website, I think that is not perfect yet but still, a good start to modernize the website.

I would like to leave a link to the working version of the changes: https://vue-test-utils-for-vue-3.netlify.app/

I am accountable so I intend to answer to every commentary and to do the alterations requested.

@lmiller1990, @afontcu what do you think of this work?

### CODE REVIEW

**Review by cexbrayat — APPROVED** [review]
The result looks good to me, thanks @nazarepiedady 

I'll let the others take a look, as I never really used vitepress

**Comment by nazarepiedady** [comment]
> The result looks good to me, thanks @nazarepiedady
> 
> I'll let the others take a look, as I never really used vitepress

Thanks @cexbrayat, so while I wait, I will resolve the conflicts.

**Review by freakzlike — APPROVED** [review]
Looks good. I like it 👍

**Comment by nazarepiedady** [comment]
@freakzlike, @cexbrayat the pull request is ready to be merged on the main branch. 😊

File: docs/.vitepress/config/shared.ts [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    // Remove control characters
+    .replace(rControl, '')
+    // Replace special characters
+    .replace(rSpecial, '-')
+    // ensure it doesn't start with a number
+    .replace(/^(\d)/, '_$1')
+
+export const sharedConfig = defineConfig({
+  title: 'Vue Test Utils',
+  appearance: 'dark',
```
Comment (lmiller1990) [inline docs/.vitepress/config/shared.ts:34]: Please default to light mode

**Review by lmiller1990 — COMMENTED** [review]
One comment.

I don't really understand what this PR accomplishes other than just updating the version of Vitepress; why is it better to have a bunch of different files for config?

File: docs/.vitepress/config/shared.ts (hunk 1 above)
Comment (nazarepiedady) [reply docs/.vitepress/config/shared.ts]: @lmiller1990 In this new version of vitepress it is only possible to set `true`, `'dark'`, and `false`, so I set it to `true` for the theme to be based on the user preference.  `false` stop the user to be able to toggle between the themes.

**Comment by nazarepiedady** [comment]
> One comment.
> 
> I don't really understand what this PR accomplishes other than just updating the version of Vitepress; why is it better to have a bunch of different files for config?

@lmiller1990 The configuration files follow the patterns observed on other projects such as Vue Router and Pinia. I think in this case different configuration files help to divide and conquer.

**Pull request merged**